OpenRA.Mods.Common/Traits/Crates/Crate.cs

```diff
@@ -91,47 +101,41 @@ public Crate(ActorInitializer init, CrateInfo info)
 				SetPosition(self, init.Get<LocationInit, CPos>());
 		}
 
-		void INotifyCrushed.WarnCrush(Actor self, Actor crusher, BitSet<CrushClass> crushClasses) { }
-
-		void INotifyCrushed.OnCrush(Actor self, Actor crusher, BitSet<CrushClass> crushClasses)
+		void CheckForCollectors(Actor self)
 		{
-			// Crate can only be crushed if it is not in the air.
-			if (!self.IsAtGroundLevel() || !crushClasses.Contains(info.CrushClass))
+			// Crate can only be collected if it is not in the air.
+			if (!self.IsAtGroundLevel())
 				return;
 
-			OnCrushInner(crusher);
-		}
+			// Check whether any other (ground) actors are in this cell.
+			var sameCellActors = self.World.ActorMap.GetActorsAt(self.Location).Where(a => a != self);
 
-		void INotifyParachute.OnParachute(Actor self) { }
-		void INotifyParachute.OnLanded(Actor self, Actor ignore)
-		{
-			// Check whether the crate landed on anything
-			var landedOn = self.World.ActorMap.GetActorsAt(self.Location)
-				.Where(a => a != self);
+			// HACK: Currently needed to find aircraft actors.
+			// TODO: Remove this once GetActorsAt supports aircraft.
+			sameCellActors.Concat(self.World.FindActorsInCircle(self.CenterPosition, new WDist(724))
```

Comment:
Couldn't we drop the `GetActorsAt` call entirely for now if we are using this afterwards anyway?

Reply:
That's what I thought as well, until I ran into everyone's favorite issue of larger buildings' CenterPosition being potentially too far away.
I guess I could use `LargestActorRadius` to fix that, in terms of performance it's probably not much of a difference anyway.

Reply:
Throwing out an idea: if we went this route we could expose the range to yaml, providing proximity collection for some usecases (e.g. proximity mines?)